### ISSUE TITLE
imagenet1k

### ISSUE THREAD

**Issue by yiqiqiandefanhua** [vimar-gu/MinimaxDiffusion#9]
torchrun --nnode=1 --master_port=25678 train_dit.py --model DiT-XL/2 \
     --data-path /nas/datasets/imagenet/train/ --ckpt pretrained_models/DiT-XL-2-256x256.pt \
     --global-batch-size 8 --tag minimax --ckpt-every 12000 --log-every 1500 --epochs 8 \
     --condense --finetune-ipc -1 --results-dir ../logs/run-0 --spec 100 --nclass 100
I used the script in the hint to achieve the accuracy of the WOOF dataset in the paper, but why does the assertion error keep happening when I implement it on imagenet, would you help me?
![Uploading 1731205342958.jpg…]()

**Comment by vimar-gu** [comment]
There is a detailed explanation in #5.

I'm sorry, but I can't see the image you uploaded. I suspect the reason is that you haven't created a corresponding class split file for `--spec 100`. 

Due to previous resource limitations, I could only run small batches. So the hyper-parameter setting might not be appropriate for large-class sets. Therefore, please use the argument `--phase` to separate the set into several sub-sets. 
The argument setting can be: `--spec 100 --nclass 20 --phase 0` for the first 20 classes.

**Comment by yiqiqiandefanhua** [comment]
> There is a detailed explanation in #5.
> 
> I'm sorry, but I can't see the image you uploaded. I suspect the reason is that you haven't created a corresponding class split file for `--spec 100`.
> 
> Due to previous resource limitations, I could only run small batches. So the hyper-parameter setting might not be appropriate for large-class sets. Therefore, please use the argument `--phase` to separate the set into several sub-sets. The argument setting can be: `--spec 100 --nclass 20 --phase 0` for the first 20 classes.

I'm sorry to bother you again, but in this way, the model parameters adjusted in the current stage will not overwrite the adjustment of the previous stage, can you help me explain?If I'm running on imagenet-1k, wouldn't I have to run the program 50 times in a row, assuming nclass=20, like
--spec 1000 --nclass 20 --phase 0
--spec 1000 --nclass 20 --phase 1
--spec 1000 --nclass 20 --phase 2
...
--spec 1000 --nclass 20 --phase 49
thanks

**Comment by vimar-gu** [comment]
Yes. The results reported on ImageNet-1K are based on 50 individual models. 

If there are sufficient resources, you can also try adjusting the hyper-parameters for larger datasets. For example, in the original Difffit paper, the authors used a batch size of 256 on 8 V100 GPU cards. In comparison, I could only use a batch size of 8. 

Highly welcome to contribute to this repo!

**Comment by yiqiqiandefanhua** [comment]
> Yes. The results reported on ImageNet-1K are based on 50 individual models. 
> 
> If there are sufficient resources, you can also try adjusting the hyper-parameters for larger datasets. For example, in the original Difffit paper, the authors used a batch size of 256 on 8 V100 GPU cards. In comparison, I could only use a batch size of 8. 
> 
> Highly welcome to contribute to this repo!

I'm sorry to bother you again, but I still don't understand. Are you referring to the fifty independent models that only use twenty classes to fine tune each model, instead of using one model to fine tune fifty times? In the second case, won't the parameters fine tuned in the previous stage be affected by the next stage? Can you help me explain
thanks

**Comment by vimar-gu** [comment]
Yes, I adopt the first setting. 50 models are parallelly fine-tuned each with 20 classes.

**Comment by yiqiqiandefanhua** [comment]
> Yes, I adopt the first setting. 50 models are parallelly fine-tuned each with 20 classes. 

Thank you for your reply. I understand now
